### PR TITLE
cleanup: remove event list

### PR DESCRIPTION
**Pull request by sanposhiho**
/kind cleanup

#### What this PR does / why we need it:

We're maintaining a huge event list in pkg/scheduler/framework/events.go, which is troublesome.
This PR aims at remove the list by creating `Label()` function, which also useful to unifying the labels for metrics.

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/127268#discussion_r1752183456

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/events.go
Comment (sanposhiho) [inline pkg/scheduler/framework/events.go:1]: Only necessary events are remaining here:
- AssignedPodXXX and UnscheduledPodXXX are necessary because we don't export action types `assignedPod` and `unschedulablePod`.
- `UnschedulableTimeout` is necessary because we have to use `label` for it.

File: pkg/scheduler/eventhandlers.go
Comment (sanposhiho) [inline pkg/scheduler/eventhandlers.go:49]: This PR is marked as draft because I wanted to create a separate PR for this change, which I found during the change for this PR.

Comment (sanposhiho) [reply pkg/scheduler/eventhandlers.go:49]: https://github.com/kubernetes/kubernetes/pull/127798

File: pkg/scheduler/framework/types.go
Comment (sanposhiho) [inline pkg/scheduler/framework/types.go:280]: We can remove this field after `UnschedulableTimeout` is removed (https://github.com/kubernetes/kubernetes/issues/87850).
Actually, `Wildcard` and `All` can also be removed at that time.

**Comment by sanposhiho** [comment]
/hold

to go thru the approver's review.

File: pkg/scheduler/framework/types.go
Comment (macsko) [inline pkg/scheduler/framework/types.go:289]:
```suggestion
	return fmt.Sprintf("%v%v", ce.Resource, ce.ActionType)
```

File: pkg/scheduler/framework/types.go
Comment (macsko) [inline pkg/scheduler/framework/types.go:304]:
```suggestion
		} else if r == Node {
```

File: pkg/scheduler/framework/types.go
Comment (macsko) [inline pkg/scheduler/framework/types.go:294]:
```suggestion
	labels := []string{UnschedulableTimeout.Label()}
```
Then we don't need line 310.

File: pkg/scheduler/framework/types.go
Comment (macsko) [reply pkg/scheduler/framework/types.go:778]: Especially see how many lines got changed because of this renaming.

File: pkg/scheduler/backend/queue/scheduling_queue.go
Comment (sanposhiho) [reply pkg/scheduler/backend/queue/scheduling_queue.go:923]: Sorry! It's a temporal log that I added for debugging test failures. Will remove it.

File: pkg/scheduler/framework/events.go
Comment (sanposhiho) [reply pkg/scheduler/framework/events.go:54]: 🤦‍♂️

File: pkg/scheduler/framework/types.go
Comment (sanposhiho) [reply pkg/scheduler/framework/types.go]: Renamed:
- `ComputeResource` -> `Resource` (= original name)
- `Resource` -> `EventResource`

File: pkg/scheduler/framework/events.go
Comment (sanposhiho) [reply pkg/scheduler/framework/events.go]: Created `unknown` ActionType to represent zero value of `ActionType` type.

**Comment by sanposhiho** [comment]
/assign @alculquicondor 

for the approval

File: pkg/scheduler/backend/queue/active_queue.go
Comment (haosdent) [inline pkg/scheduler/backend/queue/active_queue.go:307]: addEventIfPodInFlight should be addEventsIfPodInFlight

File: pkg/scheduler/framework/types.go
Comment (alculquicondor) [inline pkg/scheduler/framework/types.go:337]: so why are we making the distinction anyways? Is it just because of metrics?

File: pkg/scheduler/framework/plugins/noderesources/fit_test.go
Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/noderesources/fit_test.go:1134]: because it cannot compare a private field `label`. So, we need either `EquateComparable` or `IgnoreUnexported`

File: pkg/scheduler/framework/types.go
Comment (sanposhiho) [reply pkg/scheduler/framework/types.go:337]: Yes, just because of metrics/logs that uses `event.Label()` fn.

So, instead of creating assignedPod/unscheduledPod resources, alternatively, we can utilize `label` field in `pkg/scheduler/framework/events.go` like -

```go
// Use `Pod` with the label
AssignedPodAdd = ClusterEvent{Resource: Pod, ActionType: Add, label: "AssignedPodAdd"}
// ... do the same with all the other AssignedPod/UnschedulablePod events.
```

I didn't go with that idea just because I wouldn't like to use `label` field as much as possible, and the current implementation allows us to remove `label` field eventually, after we remove `UnschedulableTimeout` (https://github.com/kubernetes/kubernetes/issues/87850).

File: pkg/scheduler/backend/queue/active_queue.go
Comment (sanposhiho) [reply pkg/scheduler/backend/queue/active_queue.go:309]: Well, no?

File: pkg/scheduler/backend/queue/active_queue_test.go
Comment (sanposhiho) [reply pkg/scheduler/backend/queue/active_queue_test.go:50]: No specific reason.

File: pkg/scheduler/backend/queue/active_queue.go
Comment (alculquicondor) [reply pkg/scheduler/backend/queue/active_queue.go:309]: I meant on the rest of the diff, not this line specifically 😅 
But yeah, it's probably about the same.

**Comment by alculquicondor** [comment]
/approve

@haosdent, do you wanna give the LGTM?

**Comment by haosdent** [comment]
/lgtm 
Thanks @sanposhiho and @alculquicondor

**Comment by sanposhiho** [comment]
/unhold 

@alculquicondor Can you put a lgtm on it? @haosdent already did, and I just had to rebase.
This PR is conflicting with my patch on my local, so I want it to be merged soon.

**Comment by pacoxu** [comment]
/hold
feel free to unhold once the CI failure is fixed.

**Comment by sanposhiho** [comment]
Again rebased to modify the part that has broken without the conflict.

**Comment by alculquicondor** [comment]
There are still integration test failures for the scheduler

**Comment by sanposhiho** [comment]
..and caught the conflict again. Oops.

**Review by hashim21223445 — APPROVED** [review]

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Pull request merged**
